galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt

```diff
@@ -2469,6 +2553,41 @@ abstract class VBlock(var form: VForm) : VConstants, DBContextHandler, ActionHan
     TODO()
   }
 
+  /*
+   * Checks unique index constraints
+   */
+  protected fun checkUniqueIndex_(idx: Int, recno: Int, id: Int?) {
+    val buffer = mutableListOf<Op<Boolean>>()
+
+    for (field in fields) {
+
+      val col: Column<Any>? = if (field.isNull(recno) || !field.hasIndex(idx)) {
+        null
+      } else {
+        field.lookupColumn_(tables!![0])
+      }
+      if (col != null) {
+        buffer.add(Op.build { col eq field.getSql(recno)!! })
+      }
+    }
+
+    if (buffer.isNotEmpty()) {
+      transaction {
+        val query = tables!![0].slice(idColumn_).select(buffer.compoundAnd())
+
+        if (query.execute(this)!!.next()) {
+            if (query.first()[idColumn_] != id) {
+              form.setActiveBlock(this@VBlock)
+              activeRecord = recno
+              gotoFirstField()
+              throw VExecFailedException(MessageCode.getMessage("VIS-00014", arrayOf<Any>(indices!![idx])))
+            }
+          assert(!query.execute(this)!!.next()) { "too many rows" }
+        }
```

Comment:
you can optimize here by using .single() that will retrieve one element from list or throw exceptions instead